From 1abe3ae792c1116c16c5f2e8a7895f07e1baffdb Mon Sep 17 00:00:00 2001
From: WingMan Kwok <w-kwok2@ti.com>
Date: Thu, 1 Mar 2018 10:10:35 -0500
Subject: [PATCH 189/189] ethernet: ti: cpts: fix error in not unregistering
 from ptp_bc

Prior to this patch, cpts does not unregister from
the boundary clock module when cpts unregisters from
the core ptp layer. This patch makes sure it does
that.

The patch also fixes a return value error in
ptp_bc_clock_unregister (dummy version) when called
to unregister from ptp_bc module.

Signed-off-by: WingMan Kwok <w-kwok2@ti.com>
---
 drivers/net/ethernet/ti/cpts.c   | 1 +
 drivers/net/ethernet/ti/ptp_bc.h | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ti/cpts.c b/drivers/net/ethernet/ti/cpts.c
index 7d37b1b..1a68dd2 100644
--- a/drivers/net/ethernet/ti/cpts.c
+++ b/drivers/net/ethernet/ti/cpts.c
@@ -831,6 +831,7 @@ void cpts_unregister(struct cpts *cpts)
 
 	ptp_clock_unregister(cpts->clock);
 	cpts->clock = NULL;
+	ptp_bc_clock_unregister(cpts->bc_clkid);
 
 	cpts_write32(cpts, 0, int_enable);
 	cpts_write32(cpts, 0, control);
diff --git a/drivers/net/ethernet/ti/ptp_bc.h b/drivers/net/ethernet/ti/ptp_bc.h
index 1d8eadc..8d790bc 100644
--- a/drivers/net/ethernet/ti/ptp_bc.h
+++ b/drivers/net/ethernet/ti/ptp_bc.h
@@ -25,7 +25,7 @@ static int ptp_bc_clock_register(void)
 	return -1;
 }
 
-static int ptp_bc_clock_unregister(int clkid)
+static void ptp_bc_clock_unregister(int clkid)
 {
 }
 
-- 
1.9.1

